internal/commands/image/docker_client.go

```diff
@@ -58,20 +66,39 @@ func (c *Client) BuildImage(input PushImageInput, containerRepository *api.Conta
 	ctx := context.Background()
 
 	res, err := c.Cli.ImageBuild(ctx, tar, opts)
-	return &res, err
+	if err != nil {
+		return fmt.Errorf("ImageBuild: %w", err)
+	}
+	err = handleResponseBuffer(res.Body)
+	if err != nil {
+		return fmt.Errorf("HandleRespone: %w", err)
+	}
+
+	return nil
 }
 
-func (c *Client) PushImage(input PushImageInput, containerRepository *api.ContainerRepository) (io.ReadCloser, error) {
+func (c *Client) PushImage(input PushImageInput, containerRepository *api.ContainerRepository) error {
 	ctx := context.Background()
 	auth, err := createAuthForCloud(containerRepository)
-
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	res, err := c.Cli.ImagePush(ctx, imageFqn(containerRepository.RepositoryURI, input.ImageName, input.Tag), types.ImagePushOptions{RegistryAuth: auth})
+	for _, tag := range input.Tags {
+		res, err := c.Cli.ImagePush(ctx, imageFqn(containerRepository.RepositoryURI, input.ImageName, tag), types.ImagePushOptions{RegistryAuth: auth})
```

Comment:
The docker client doesnt allow us to push the same image with multiple tags?

Reply:
Kinda, there is an `All           bool` option but the problem with that is then every image that matched would get pushed even if that really isn't the intention. This is the safer way since push operates on more of a single image and once the image is up, it's fast to `push` another tag.

Reply:
sounds good.